### DISCUSSION TITLE
[Feature Request] Prompt before deleting images when deleting a stack

### DISCUSSION THREAD

**Discussion by anthonysmillie** [louislam/dockge#25] [Feature Request]
One of the images on my local is one I built myself quite some time ago, and no longer have the build files for. Thankfully I didn't end up losing the image, but I noticed when deleting one of my other stacks that dockge deletes the associated image files. 

I know this is an edge case, as most people will be using standard images from a registry, and honestly it's my own fault that I no longer have the build files available, but I'd really appreciate if not deleting images was either a setting, or at the very least a prompt asking yes/no to deleting images associated with a stack when deleting a stack.

**Comment by louislam** [comment]
Good point, because I added `--rmi` (Remove image) in the `docker compose down` command.

I will remove `--rmi`, it is actually not safe


https://github.com/louislam/dockge/blob/96a4f2fd0ca206de4b6a8eb408343526d132b0d1/backend/stack.ts#L152C79-L152C144

**Reply by Tivin-i** [reply]
I'd propose to prompt a stack deletion ("type _stack name_ to delete") and maybe have a checklist where you can choose if to remove volumes/folders/images.

**Reply by EricChen1248** [reply]
Upvote for the prompt.
I fully admit this is my fault. I had thought the delete button was delete the images that were spun up (a docker compose down), not delete everything related to that compose stack including folders.

I had database and application data stored as subdirs in the same folder. Delete just wiped them all.

Luckily this happened on my gitea instance, where the only thing of value lost were some minor issues I was tracking for myself. Every repo had a local copy that I could just push back (thank goodness git is distributed model)

But having a prompt saying what that button would have done would have avoided all this.
